src/components/Counter/Counter.jsx

```diff
@@ -7,51 +8,66 @@ const colors = {
   red: 'rgb(239, 68, 68)',
 };
 
+const initialCounter = { count: 0, currentColor: colors.yellow };
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'increment':
+      return { count: state.count + 1, currentColor: colors.yellow };
+    case 'decrement':
+      return { count: state.count - 1, currentColor: colors.red };
+    case 'reset':
+      return { count: (state.count = 0), currentColor: colors.yellow };
```

Comment:
We can spread in the state here => { ...state, count: state.count... } since we aren't manipulating the colors in these actions.